recipes/lora_finetune_distributed.py

```diff
@@ -307,16 +303,40 @@ def _setup_model(
                 base_model_state_dict_keys=base_model_state_dict.keys(),
             )
 
-            # Load both the base model weights and (if available) the adapter weights. Both
-            # of this should happen only on Rank 0
-            model.load_state_dict(base_model_state_dict, strict=False)
-            if lora_weights_state_dict:
-                model.load_state_dict(lora_weights_state_dict, strict=False)
+        self.adapter_params = get_adapter_params(model)
+        set_trainable_params(model, self.adapter_params)
 
-        else:
-            # For non-zero ranks, load the model on meta device
-            with utils.set_default_dtype(self._dtype), torch.device("meta"):
-                model = config.instantiate(cfg_model)
+        if enable_activation_checkpointing:
+            utils.set_activation_checkpointing(
+                model, auto_wrap_policy={modules.TransformerDecoderLayer}
+            )
+
+        for m in reversed(list(model.modules())):
+            if (
+                isinstance(m, nn.Linear)
```

Comment:
Does this need to be LoRALinear? Or am I misunderstanding?

Reply:
`nn.Linear` with `required_grad=True` are referring to `lora_a` and `lora_b`

Trunk FSDP1 is wrapping `lora_a` and `lora_b` separately thus we are do the same wrapping for parity. But as you mentioned, we can wrap `LoRALinear` instead so `lora_a` and `lora_b` are communicated together